### PR TITLE
Add j-tock auth

### PR DESCRIPTION
**Pull request by dorianbuck**
PT: https://www.pivotaltracker.com/story/show/180092183

## User Story
```
As a User
In order to sign to my registered account
I would like to have the functionality to do so
```
## Tasks
- Install J-tockauth 
- Use j-tock auth sign up function
### ScreenShots

Skärmavbild 2021-10-27 kl  16 31 23

### CODE REVIEW

File: src/components/Register.jsx [hunk 1]
```diff
@@ -9,23 +9,40 @@ import {
...
+    let response = await auth
+      .signUp(email, password)
+      .then((response) => {
+        if (response.data.status === "success") {
+          setOpen(true);
+          dispatch({
+            type: "SET_CURRENT_USER",
+            payload: response.data,
+          });
+        } else {
```
Comment (Fraseer) [inline src/components/Register.jsx]: We can remove this else block

File: src/components/Register.jsx [hunk 2]
```diff
@@ -9,23 +9,40 @@ import {
...
-  const handleSubmit = (event) => {
-    Authentication.register(event).then(setOpen(true));
-    document.getElementById("form-input-control-error-email").value = "";
-    document.getElementById("form-input-password").value = "";
-    document.getElementById("form-input-confirm-password").value = "";
+  const handleAuthentication = async () => {
+    let response = await auth
+      .signUp(email, password)
+      .then((response) => {
+        if (response.data.status === "success") {
```
Comment (Fraseer) [inline src/components/Register.jsx]: And maybe the if as well

File: src/modules/auth.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import JtockAuth from "j-tockauth";
+
+const auth = new JtockAuth({
+  host: "",
```
Comment (Fraseer) [inline src/modules/auth.js]: Does this do anything if it's an empty string?

**Review by Fraseer — COMMENTED** [review]
Looks good just a couple small things and the deploy

File: src/modules/auth.js (hunk 3 above)
Comment (dorianbuck) [reply src/modules/auth.js]: I tried removing them but it didn't like it

File: src/components/Register.jsx (hunk 1 above)
Comment (dorianbuck) [reply src/components/Register.jsx]: Done!

File: cypress/integration/userCanRegisterForAnAccount.feature.js [hunk 4]
```diff
@@ -22,6 +29,7 @@ describe("User can register for an account", () => {
       cy.get("[data-cy=btn-signup]").click();
     });
     it("is expected to display a success message", () => {
+
```
Comment (maxarvid) [inline cypress/integration/userCanRegisterForAnAccount.feature.js]: Delete empty line here?

File: src/modules/auth.js (hunk 3 above)
Comment (lukeperrin10) [reply src/modules/auth.js]: does this work being blank? how does it know where to make the call? Have you tried manually testing this?

Comment (dorianbuck) [reply src/modules/auth.js]: We are setting the baseURL in index.js. I haven't tested it but the address is correct in dev tools. Is that enough?

Comment (lukeperrin10) [reply src/modules/auth.js]: so when you spin this code up locally it points to the API and you're able to sign in etc?

File: src/components/Register.jsx [hunk 5]
```diff
@@ -9,42 +9,66 @@ import {
...
-    Authentication.register(event).then(setOpen(true));
-    document.getElementById("form-input-control-error-email").value = "";
-    document.getElementById("form-input-password").value = "";
-    document.getElementById("form-input-confirm-password").value = "";
+  const handleAuthentication = async (event) => {
+    event.preventDefault();
+    const form = event.target;
+    const email = form.email.value;
+    const password = form.password.value;
+    const confirmPassword = form.confirmPassword.value;
```
Comment (maxarvid) [inline src/components/Register.jsx:27]: Consider adding a `const name` here since the input was left in below

File: src/modules/auth.js (hunk 3 above)
Comment (dorianbuck) [reply src/modules/auth.js]: There was an issue with the address when spun up locally. This has now been adjusted to work properly. Thanks!

**Review by Fraseer — APPROVED** [review]

File: cypress/integration/userCanRegisterForAnAccount.feature.js [hunk 6]
```diff
@@ -3,19 +3,40 @@ describe("User can register for an account", () => {
     cy.intercept("GET", "**api/articles", {
       fixture: "indexRespondsFromApi.json",
     });
+    cy.intercept("POST", "**api/auth/sign_in", {
+      fixture: "authenticationSuccess.json",
+      headers: {
+        "access-token": "abcd1dMVlvW2BT67xIAS_A",
+        "token-type": "Bearer",
+        client: "LSJEVZ7Pq6DX5LXvOWMq1w",
+        expiry: "1519086891",
```
Comment (lukeperrin10) [inline cypress/integration/userCanRegisterForAnAccount.feature.js:12]: you shouldn't need all this for the test, it should be enough with `headers: {uid: 'user@mail.com'}`. No need to change it for this one but check it out going forward

File: src/modules/auth.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import JtockAuth from "j-tockauth";
+
+let url = "";
+if (!process.env.NODE_ENV || process.env.NODE_ENV === "development") {
+  url = "http://localhost:3000";
+} else {
+  url = "https://news-in-progress-api.herokuapp.com";
+}
```
Comment (lukeperrin10) [inline src/modules/auth.js:8]: Great!

**Review by lukeperrin10 — APPROVED** [review]
Good work!

**Pull request merged**